geoscilabs/dcip/DCWidgetPlate2_5D.py

```diff
@@ -53,51 +55,77 @@
     & (mesh.gridFy[:, 1] <= ymax)
 )
 indF = np.concatenate((indx, indy))
+_cache = {
+    "A": None,
+    "B": None,
+    "dx": None,
+    "dz": None,
+    "xc": None,
+    "zc": None,
+    "rotAng": None,
+    "sigplate": None,
+    "sighalf": None,
+}
 
 
 def plate_fields(A, B, dx, dz, xc, zc, rotAng, sigplate, sighalf):
-    # Create halfspace model
-    mhalf = np.log(sighalf * np.ones([mesh.nC]))
-    # mhalf = sighalf*np.ones([mesh.nC,])
-
-    # Create true model with plate
-    mtrue = createPlateMod(xc, zc, dx, dz, rotAng, sigplate, sighalf)
-
-    Mx = mesh.gridCC
-    # Nx = np.empty(shape=(mesh.nC, 2))
-    rx = DC.receivers.Pole_ky(Mx)
-    # rx = DC.receivers.Dipole(Mx,Nx)
-    if B == []:
-        src = DC.sources.Pole([rx], np.r_[A, 0.0])
+    re_run = (
+        _cache["A"] != A
+        or _cache["B"] != B
+        or _cache["dx"] != dx
+        or _cache["dz"] != dz
+        or _cache["xc"] != xc
+        or _cache["zc"] != zc
+        or _cache["rotAng"] != rotAng
+        or _cache["sigplate"] != sigplate
+        or _cache["sighalf"] != sighalf
+    )
+    if re_run:
+        # Create halfspace model
+        mhalf = np.log(sighalf * np.ones([mesh.nC]))
+        # mhalf = sighalf*np.ones([mesh.nC,])
```

Comment:
do you think we still need these (and similarly on 94, etc)? If not, we could clear them out